oci/private/tarball.sh.tpl

```diff
@@ -1,37 +1,140 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit -o nounset
 
+readonly FORMAT="{{format}}"
 readonly STAGING_DIR=$(mktemp -d)
 readonly YQ="{{yq}}"
 readonly IMAGE_DIR="{{image_dir}}"
 readonly BLOBS_DIR="${STAGING_DIR}/blobs"
 readonly TARBALL_PATH="{{tarball_path}}"
 readonly TAGS_FILE="{{tags}}"
+readonly INDEX_FILE="${IMAGE_DIR}/index.json"
+
+cp_f_with_mkdir() {
+  SRC="$1"
+  DST="$2"
+  mkdir -p "$(dirname "${DST}")"
+  cp -f "${SRC}" "${DST}"
+}
+
+MANIFEST_DIGEST=$(${YQ} eval '.manifests[0].digest | sub(":"; "/")' "${INDEX_FILE}" | tr  -d '"')
+REPOTAGS=($(cat "${TAGS_FILE}"))
+
+MANIFESTS_LENGTH=$("${YQ}" eval '.manifests | length' "${INDEX_FILE}")
+if [[ "${MANIFESTS_LENGTH}" != 1 ]]; then
+  echo >&2 "Expected exactly one manifest in ${INDEX_FILE}"
+  exit 1
+fi
+
+MEDIA_TYPE=$("${YQ}" eval ".manifests[0].mediaType" "${INDEX_FILE}")
+
+# Check that we know how to generate the output format given the input format.
+# We may expand the supported options here in the future, but for now,
+if [[ "${FORMAT}" != "docker" && "${FORMAT}" != "oci" ]]; then
+  echo >&2 "Unknown format: ${FORMAT}. Only support docker|oci"
+  exit 1
+fi
+if [[ "${FORMAT}" == "oci" && "${MEDIA_TYPE}" != "application/vnd.oci.image.index.v1+json" ]]; then
```

Comment:
Can we also allow docker manifests lists here? i don't see a reason why we should restrict this.

Reply:
We can! Now we do!